sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -184,15 +189,62 @@ abstract class JdbcDialect extends Serializable {
   /**
    * Rename an existing table.
    *
-   * TODO (SPARK-32382): Override this method in the dialects that don't support such syntax.
-   *
    * @param oldTable The existing table.
    * @param newTable New name of the table.
    * @return The SQL statement to use for renaming the table.
    */
   def renameTable(oldTable: String, newTable: String): String = {
     s"ALTER TABLE $oldTable RENAME TO $newTable"
   }
+
+  /**
+   * Alter an existing table.
+   * TODO (SPARK-32523): Override this method in the dialects that have different syntax.
+   *
+   * @param tableName The name of the table to be altered.
+   * @param changes Changes to apply to the table.
+   * @return The SQL statements to use for altering the table.
+   */
+  def alterTable(tableName: String, changes: Seq[TableChange]): Array[String] = {
+    val updateClause = ArrayBuilder.make[String]
+    for (change <- changes) {
+      change match {
+        case add: AddColumn =>
+          add.fieldNames match {
+            case Array(name) =>
```

Comment:
we should have a better error message if the field name has more than one parts.

Reply:
Seems to me fieldName always has only one element.

Reply:
If we do believe it always has one element, maybe add an assert?

Reply:
People can alter a nested field, that why the type of `fieldNames` is `Array[String]`. e.g. `ALTER TABLE t RENAME COLUMN a.b TO c`

Reply:
Does this look OK?
```
            case _ =>
              throw new SQLFeatureNotSupportedException("Nested column is not supported.")
```

Reply:
```
add.fieldNames match {
            case Array(name) =>
···
We will fail at this pattern match.

One way is
```
case add: AddColumn if add.fieldNames.length == 1 =>
```

Reply:
@cloud-fan I changed code to what you suggested.
We don't support nested column in JDBC yet. In `JdbcUtils.getCatalystType`, we map `java.sql.Types.STRUCT        => StringType`. In `getCommonJDBCType`, we don't  have a match case for StructType and the default is None. Seems to me currently it is not possible to reach the code path of multiple parts fieldNames in `JdbcDialects.alterTable`, so I will not have a negative test case for this code path for now.